src/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegment.cs

```diff
@@ -79,6 +79,8 @@ public void ResetMemory()
             AvailableMemory = default;
         }
 
+        internal OwnedMemory<byte> OwnedMemory => _ownedMemory;
```

Comment:
You could've used AvailableMemory == default instead exposing OM

Reply:
How?